File: backend/models/Service.js
const mongoose = require('mongoose');

const serviceSchema = new mongoose.Schema({
  title: {
    type: String,
    required: [true, 'Service title is required'],
    trim: true
  },
  description: {
    type: String,
    required: [true, 'Service description is required']
  },
  category: {
    type: String,
    required: [true, 'Category is required'],
    enum: [
      'web-development',
      'mobile-development',
      'ui-ux-design',
      'graphic-design',
      'digital-marketing',
      'seo',
      'content-writing',
      'video-editing',
      'data-analysis',
      'consulting'
    ]
  },
  price: {
    type: Number,
    required: [true, 'Price is required'],
    min: 0
  },
  deliveryTime: {
    type: String,
    required: [true, 'Delivery time is required']
  },
  contactInfo: {
    type: String,
    required: false
  },
  image: {
    type: String,
    default: 'https://images.unsplash.com/photo-1498050108023-c5249f4df085?w=400'
  },
  images: {
    type: [String],
    default: []
  },
  freelancerId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  createdBy: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  rating: {
    type: Number,
    default: 0,
    min: 0,
    max: 5
  },
  reviews: [{
    clientId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true
    },
    comment: {
      type: String,
      required: true
    },
    stars: {
      type: Number,
      required: true,
      min: 1,
      max: 5
    },
    createdAt: {
      type: Date,
      default: Date.now
    }
  }],
  tags: [String],
  isActive: {
    type: Boolean,
    default: true
  },
  views: {
    type: Number,
    default: 0
  },
  orders: {
    type: Number,
    default: 0
  }
}, {
  timestamps: true
});

// Index for better search performance
serviceSchema.index({ title: 'text', description: 'text', tags: 'text' });
serviceSchema.index({ category: 1 });
serviceSchema.index({ createdBy: 1 });

module.exports = mongoose.model('Service', serviceSchema);

File: freelancex-app/app/src/main/java/com/freelancex/data/model/Service.kt
package com.freelancex.data.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

/**
 * Service data model matching the backend MongoDB schema
 */
@Parcelize
data class Service(
    @SerializedName("_id")
    val id: String = "",
    
    @SerializedName("title")
    val title: String = "",
    
    @SerializedName("description")
    val description: String = "",
    
    @SerializedName("category")
    val category: String = "",
    
    @SerializedName("price")
    val price: Double = 0.0,
    
    @SerializedName("deliveryTime")
    val deliveryTime: String = "",
    
    @SerializedName("image")
    val image: String? = null,
    
    @SerializedName("images")
    val images: List<String> = emptyList(),
    
    @SerializedName("tags")
    val tags: List<String> = emptyList(),
    
    @SerializedName("rating")
    val rating: Double = 0.0,
    
    @SerializedName("reviewCount")
    val reviewCount: Int = 0,
    
    @SerializedName("orderCount")
    val orderCount: Int = 0,
    
    @SerializedName("createdBy")
    val createdBy: User? = null,
    
    @SerializedName("freelancerId")
    val freelancerId: String? = null, // Alternative field name
    
    @SerializedName("isActive")
    val isActive: Boolean = true,
    
    @SerializedName("featured")
    val featured: Boolean = false,
    
    @SerializedName("createdAt")
    val createdAt: String = "",
    
    @SerializedName("updatedAt")
    val updatedAt: String = ""
) : Parcelable

/**
 * Service category enum for type safety
 */
enum class ServiceCategory(val value: String, val displayName: String, val icon: String) {
    WEB_DEVELOPMENT("web-development", "Web Development", "💻"),
    MOBILE_DEVELOPMENT("mobile-development", "Mobile Development", "📱"),
    UI_UX_DESIGN("ui-ux-design", "UI/UX Design", "🎨"),
    GRAPHIC_DESIGN("graphic-design", "Graphic Design", "🖼️"),
    DIGITAL_MARKETING("digital-marketing", "Digital Marketing", "📈"),
    SEO("seo", "SEO", "🔍"),
    CONTENT_WRITING("content-writing", "Content Writing", "✍️"),
    VIDEO_EDITING("video-editing", "Video Editing", "🎬"),
    DATA_ANALYSIS("data-analysis", "Data Analysis", "📊"),
    CONSULTING("consulting", "Consulting", "💼");
    
    companion object {
        fun fromValue(value: String): ServiceCategory? {
            return values().find { it.value == value }
        }
        
        fun getAllCategories(): List<ServiceCategory> = values().toList()
    }
}

/**
 * Extension functions for Service model
 */
fun Service.getFreelancerId(): String? = createdBy?.id ?: freelancerId

fun Service.getMainImage(): String? = image ?: images.firstOrNull()

fun Service.getCategoryEnum(): ServiceCategory? = ServiceCategory.fromValue(category)

fun Service.getFormattedPrice(): String = "$${String.format("%.2f", price)}"

fun Service.getRatingText(): String = if (reviewCount > 0) {
    "${String.format("%.1f", rating)} (${reviewCount} reviews)"
} else {
    "No reviews yet"
}